opm/simulators/wells/WellState.hpp

```diff
@@ -152,13 +152,28 @@ namespace Opm
             return wellRates().size() / numWells();
         }
 
+
+        void shutWell(int well_index) {
+            this->open_for_output_[well_index] = false;
+            this->thp_[well_index] = 0;
+            this->bhp_[well_index] = 0;
+            const int np = numPhases();
+            for (int p = 0; p < np; ++p) {
+                this->wellrates_[np * well_index + p] = 0;
+                this->perfrates_[np * well_index + p] = 0;
```

Comment:
What's the size of `perfrates_`?  This looks to be treating it as an `np`-per-well quantity, but from the name I would expect `np`-per-reservoir connection size instead.

Reply:
I agree that looks suspicious; I have just copied the code verbatim from the original location in `BlackoilWellModel_impl.hpp` - but it might have been incorrect there as well.